File: domus-plus-codigo-completo/backend/app/services/whatsapp_service.py
import os
from typing import Optional

try:
    from twilio.rest import Client
    from twilio.twiml.messaging_response import MessagingResponse
    TWILIO_AVAILABLE = True
except ImportError:
    TWILIO_AVAILABLE = False
    Client = None
    MessagingResponse = None

account_sid = os.getenv("TWILIO_ACCOUNT_SID")
auth_token = os.getenv("TWILIO_AUTH_TOKEN")
whatsapp_number = os.getenv("TWILIO_WHATSAPP_NUMBER")

client = Client(account_sid, auth_token) if (TWILIO_AVAILABLE and account_sid and auth_token) else None

def send_whatsapp_message(to: str, message: str) -> bool:
    """
    Envía un mensaje de WhatsApp usando Twilio.
    """
    if not client:
        print("Twilio no configurado")
        return False
    
    try:
        message = client.messages.create(
            body=message,
            from_=whatsapp_number,
            to=f"whatsapp:{to}"
        )
        return True
    except Exception as e:
        print(f"Error enviando mensaje WhatsApp: {str(e)}")
        return False

def process_incoming_message(phone: str, message_body: str, media_url: Optional[str] = None) -> dict:
    """
    Procesa un mensaje entrante de WhatsApp.
    Retorna información sobre el tipo de mensaje.
    """
    return {
        "phone": phone,
        "message_body": message_body,
        "has_media": media_url is not None,
        "media_url": media_url
    }

File: lib/services/whatsapp-service.ts
/**
 * Servicio para WhatsApp usando Twilio
 * Requiere: npm install twilio
 */

function getTwilioClient() {
  try {
    const twilio = require('twilio')
    const accountSid = process.env.TWILIO_ACCOUNT_SID
    const authToken = process.env.TWILIO_AUTH_TOKEN
    
    if (!accountSid || !authToken) {
      return null
    }
    
    return twilio(accountSid, authToken)
  } catch (error) {
    console.error('Twilio no disponible:', error)
    return null
  }
}

export function sendWhatsAppMessage(to: string, message: string): boolean {
  const client = getTwilioClient()
  if (!client) {
    console.log('Twilio no configurado')
    return false
  }

  const whatsappNumber = process.env.TWILIO_WHATSAPP_NUMBER
  if (!whatsappNumber) {
    console.log('TWILIO_WHATSAPP_NUMBER no configurado')
    return false
  }

  try {
    client.messages.create({
      body: message,
      from: whatsappNumber,
      to: `whatsapp:${to}`,
    })
    return true
  } catch (error: any) {
    console.error(`Error enviando mensaje WhatsApp: ${error.message}`)
    return false
  }
}
